### PR TITLE
Kernel: PCI host bridge fixes before implementing VMD support

### PR DESCRIPTION
**Pull request by supercomputer7**
Another preparation before completely fixing #11594.

### CODE REVIEW

**Comment by supercomputer7** [comment]
A basic implementation for the intel vmd bridge exists here:
https://github.com/supercomputer7/serenity/tree/intel-vmd
I'm waiting to work on this after this pull request is merged, so I can ensure it's working correctly on @tomuta machine.

File: Kernel/Bus/PCI/Controller/HostBridge.cpp [hunk 1]
```diff
@@ -53,7 +53,7 @@ UNMAP_AFTER_INIT Vector<Capability> HostBridge::get_capabilities_for_function(Bu
 UNMAP_AFTER_INIT void HostBridge::enumerate_functions(Function<void(DeviceIdentifier)> const& callback, BusNumber bus, DeviceNumber device, FunctionNumber function, bool recursive_search_into_bridges)
 {
     dbgln_if(PCI_DEBUG, "PCI: Enumerating function, bus={}, device={}, function={}", bus, device, function);
-    Address address(0, bus.value(), device.value(), function.value());
```
Comment (IdanHo) [inline Kernel/Bus/PCI/Controller/HostBridge.cpp:56]: The title of this commit seems a bit confusing, Do you mean "Don't hardcode address domain to 0 when enumerating devices"?

File: Kernel/Bus/PCI/Controller/HostBridge.cpp [hunk 2]
```diff
@@ -106,8 +106,8 @@ UNMAP_AFTER_INIT void HostBridge::enumerate_attached_devices(Function<void(Devic
     VERIFY(Access::the().scan_lock().is_locked());
     // First scan bus 0. Find any device on that bus, and if it's a PCI-to-PCI
     // bridge, recursively scan it too.
```
Comment (IdanHo) [inline Kernel/Bus/PCI/Controller/HostBridge.cpp:108]: `might put` => `might have`

**Review by IdanHo — COMMENTED** [review]
Looks fine, just a couple of commit message nitpicks

**Review by IdanHo — APPROVED** [review]

**Pull request merged**